examples/window.rs

```diff
@@ -1,43 +1,980 @@
+                    .is_pressed()
+                    .then(|| Self::process_mouse_binding(button, &mods))
+                    .flatten()
+                {
+                    self.handle_action(event_loop, window_id, action);
+                }
+            }
+            WindowEvent::CursorLeft { .. } => {
+                println!("Cursor left Window={window_id:?}");
+                window.cursor_left();
+            }
+            WindowEvent::CursorMoved { position, .. } => {
+                println!("Moved cursor to {position:?}");
+                window.cursor_moved(position);
+            }
+            WindowEvent::ActivationTokenDone { token: _token, .. } => {
+                #[cfg(any(x11_platform, wayland_platform))]
+                {
+                    startup_notify::set_activation_token_env(_token);
+                    if let Err(err) = self.create_window(event_loop, None) {
+                        eprintln!("Error creating new window: {err}");
+                    }
+                }
+            }
+            WindowEvent::Ime(event) => match event {
+                Ime::Enabled => println!("IME enabled for Window={window_id:?}"),
+                Ime::Preedit(text, caret_pos) => {
+                    println!("Preedit: {}, with caret at {:?}", text, caret_pos);
+                }
+                Ime::Commit(text) => {
+                    println!("Commited: {}", text);
+                }
+                Ime::Disabled => println!("IME disabled for Window={window_id:?}"),
             },
-            Event::AboutToWait => {
-                window.request_redraw();
+            WindowEvent::PinchGesture { delta, .. } => {
+                window.zoom += delta;
+                let zoom = window.zoom;
+                if delta > 0.0 {
+                    println!("Zoomed in {delta:.5} (now: {zoom:.5})");
+                } else {
+                    println!("Zoomed out {delta:.5} (now: {zoom:.5})");
+                }
+            }
+            WindowEvent::RotationGesture { delta, .. } => {
+                window.rotated += delta;
+                let rotated = window.rotated;
+                if delta > 0.0 {
+                    println!("Rotated counterclockwise {delta:.5} (now: {rotated:.5})");
+                } else {
+                    println!("Rotated clockwise {delta:.5} (now: {rotated:.5})");
+                }
             }
+            WindowEvent::DoubleTapGesture { .. } => {
+                println!("Smart zoom");
+            }
+            WindowEvent::TouchpadPressure { .. }
+            | WindowEvent::HoveredFileCancelled
+            | WindowEvent::KeyboardInput { .. }
+            | WindowEvent::CursorEntered { .. }
+            | WindowEvent::AxisMotion { .. }
+            | WindowEvent::DroppedFile(_)
+            | WindowEvent::HoveredFile(_)
+            | WindowEvent::Destroyed
+            | WindowEvent::Touch(_)
+            | WindowEvent::Moved(_) => (),
```

Comment:
Nit: I'm not sure we should encourage exhaustive matching? It makes it more of a breaking change to add a variant.
```suggestion
            _ => {},
```

Reply:
it's mostly to show things we have, don't see why not?

Reply:
I think I wanted to make `WindowEvent` be `#[non_exhaustive]`, but since we don't have that, I guess it doesn't matter that much.

Reply:
Given that `WindowEvent` will be likely gone or drastically reworked in 0.31 marking it non_exhaustive sounds like not really a great thing for 0.30.